File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/routes-compiler-routes-compilation-java/app/controllers/InstanceController.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package controllers;

import jakarta.inject.Inject;
import play.mvc.Controller;
import play.mvc.Result;

public class InstanceController extends Controller {

  @Inject
  public InstanceController() {}

  public Result index() {
    return ok();
  }
}

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/routes-compiler-routes-compilation/app/controllers/InstanceController.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package controllers

import jakarta.inject.Inject
import play.api.mvc._

class InstanceController @Inject() (c: ControllerComponents) extends AbstractController(c) {
  def index = Action {
    Ok
  }
}
